### PR TITLE
11 add rlhf soft actor critic playground implementation

### PR DESCRIPTION
**Pull request by bkuen**
This is a really basic version of an RLHF framework based on the CleanRL SAC implementation.
I am not sure whether the implementation already works. There is an improvement in the rewards especially in the beginning, but I can observe some oscillating behaviour at the end. There are also some issues caused by "overflowing" buffers. 

However, I think the main idea and structure behind this code is a good starting point for the development of our framework.

### CODE REVIEW

File: framework/preference_buffer.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        self.size = 0
+        self.next_idx = 0
+
+    def add(self, first_trajectory: Trajectory, second_trajectory: Trajectory, preference: float):
+        """
+        Adds a new preference to the buffer. If the buffer is full, it overwrites the oldest entry.
+
+        :param first_trajectory: First trajectory being compared
+        :param second_trajectory: Second trajectory being compared
+        :param preference: Preference score (e.g., 1 if first is preferred, -1 if second is preferred, 0 for equal preference)
```
Comment (Wizzardexe) [inline framework/preference_buffer.py:18]: Is there a reason that the preference values returned by give_preference (2.0, 0.5, 1.0) are not aligned with the expected values in the add method (-1, 0, 1)?

Comment (Wizzardexe) [reply framework/preference_buffer.py:18]: I think using (-1, 0, 1) for preferences as it aligns better with our preference_loss() implementation. This format ensures that the loss calculation (-torch.sum(preferences * torch.log(predictions))) operates as intended, with -1 representing a preference for the second option, 1 for the first option, and 0 for equal preference. Using any other format, such as (2.0, 1.0, 0.5), would require additional preprocessing and could lead to inconsistencies in the loss calculation.

Comment (Wizzardexe) [reply framework/preference_buffer.py:18]: I just noticed in the PEBBLE paper they use (0, 0.5, 1) and the loss looks like Binary Cross-Entropy.

**Review by Wizzardexe — APPROVED** [review]
The implementations provided are a solid foundation to build upon. In our meeting tomorrow, let's explore options to optimize these further and outline a plan to divide the tasks among the team for implementation.
